instaclustr/api_mock_client.go

```diff
@@ -1,14 +1,12 @@
-package test
+package instaclustr
 
 import (
 	"net/http"
 	"time"
-
-	"github.com/instaclustr/terraform-provider-instaclustr/instaclustr"
 )
 
 type APIMockClient struct {
-	instaclustr.APIClient
+	APIClient
```

Comment:
**[Desired, Optimisation]** Currently, this will pass the `APIClient` struct on the heap and store it in an instance of this struct. That's not great. Instead, we should pass a pointer to an APIClient to this structure as that avoids stack memory restructuring and also eliminates the overhead with memory drop with context changes

Reply:
Discussed with Hail, will remove the APIMockClient struct and use a function to return a mocked API client